pkg/varlinkapi/containers_create.go

```diff
@@ -0,0 +1,243 @@
+package varlinkapi
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+	"syscall"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/pkg/signal"
+	"github.com/projectatomic/libpod/cmd/podman/libpodruntime"
+	"github.com/projectatomic/libpod/cmd/podman/varlink"
+	"github.com/projectatomic/libpod/libpod"
+	"github.com/projectatomic/libpod/libpod/image"
+	"github.com/projectatomic/libpod/pkg/inspect"
+	cc "github.com/projectatomic/libpod/pkg/spec"
+	"github.com/projectatomic/libpod/pkg/util"
+)
+
+// CreateContainer ...
+func (i *LibpodAPI) CreateContainer(call ioprojectatomicpodman.VarlinkCall, config ioprojectatomicpodman.Create) error {
+	//mappings, err := util.ParseIDMapping(config.Uidmap, config.Gidmap, config.Subuidmap, config.Subgidmap)
+	//if err != nil {
+	//	return err
+	//}
+	//storageOpts := storage.DefaultStoreOptions
+	//storageOpts.UIDMap = mappings.UIDMap
+	//storageOpts.GIDMap = mappings.GIDMap
+
+	runtime, err := libpodruntime.GetRuntime(i.Cli)
+	if err != nil {
+		return call.ReplyRuntimeError(err.Error())
+	}
+	defer runtime.Shutdown(false)
+
+	rtc := runtime.GetConfig()
+	ctx := getContext()
+
+	newImage, err := runtime.ImageRuntime().New(ctx, config.Image, rtc.SignaturePolicyPath, "", os.Stderr, nil, image.SigningOptions{}, false, false)
+	if err != nil {
+		return err
+	}
+	data, err := newImage.Inspect(ctx)
+
+	createConfig, err := varlinkCreateToCreateConfig(ctx, config, runtime, config.Image, data)
+	if err != nil {
+		return call.ReplyErrorOccurred(err.Error())
+	}
+	useImageVolumes := createConfig.ImageVolumeType == "bind"
+
+	runtimeSpec, err := cc.CreateConfigToOCISpec(createConfig)
+	if err != nil {
+		return call.ReplyErrorOccurred(err.Error())
+	}
+
+	options, err := createConfig.GetContainerCreateOptions()
+	if err != nil {
+		return call.ReplyErrorOccurred(err.Error())
+	}
+	// Gather up the options for NewContainer which consist of With... funcs
+	options = append(options, libpod.WithRootFSFromImage(createConfig.ImageID, createConfig.Image, useImageVolumes))
```

Comment:
We should move all these into `createConfig.GetContainerCreateOptions()` so we can share code between this, run, and create

Reply:
can we defer that to another PR; given the huge change, i really wanted to achieve getting this working first.

Reply:
deferred